crates/redeem/src/lib.rs

```diff
@@ -325,21 +325,21 @@ impl<T: Config> Pallet<T> {
             .checked_sub(&fee_wrapped)
             .ok_or(Error::<T>::ArithmeticUnderflow)?;
 
-        // this can overflow for small requested values. As such return AmountBelowDustAmount when this happens
-        let user_to_be_received_btc = vault_to_be_burned_tokens
-            .checked_sub(&inclusion_fee)
-            .ok_or(Error::<T>::AmountBelowDustAmount)?;
-
-        ext::vault_registry::ensure_not_banned::<T>(&vault_id)?;
-
         // only allow requests of amount above above the minimum
         let dust_value = <RedeemBtcDustValue<T>>::get();
         ensure!(
             // this is the amount the vault will send (minus fee)
-            user_to_be_received_btc >= dust_value,
+            dust_value + inclusion_fee <= vault_to_be_burned_tokens,
```

Comment:
Imo the previous version is more clear - the amount of bitcoin to be transferred must be above the dust amount. But I guess it's a matter of preference. The previous version does save one calculation though. (and besides that, don't use unchecked math in parachain code).